Marlin/src/HAL/HAL_DUE/HAL_timers_Due.h

```diff
@@ -49,7 +49,7 @@
 #define HAL_TIMER_RATE         ((F_CPU) / 2.0)  // frequency of timers peripherals
 #define STEPPER_TIMER_PRESCALE 1.0              // prescaler for setting stepper frequency
 #define HAL_STEPPER_TIMER_RATE HAL_TIMER_RATE   // frequency of stepper timer (HAL_TIMER_RATE / STEPPER_TIMER_PRESCALE)
-#define HAL_TICKS_PER_US       ((HAL_STEPPER_TIMER_RATE) / 1000000) // stepper timer ticks per us
+#define HAL_TICKS_PER_US       (((F_CPU) / 2) / 1000000) // stepper timer ticks per us, can not be of type double
```

Comment:
What if `HAL_TIMER_RATE` was defined as `((F_CPU) / 2)` instead?

Reply:
That was my first thought, but I didn't have time to go through and check if bobc used the type promotion to double anywhere with all these defines.. you don't general put .0 unless its for a reason.

Reply:
Seems like it would be safe, given the `calc_timer` code and others…
```cpp
        // In case of high-performance processor, it is able to calculate in real-time
        constexpr uint32_t MIN_TIME_PER_STEP = HAL_STEPPER_TIMER_RATE / (STEP_DOUBLER_FREQUENCY * 2);
        timer = (uint32_t)(HAL_STEPPER_TIMER_RATE) / step_rate;
        NOLESS(timer, MIN_TIME_PER_STEP); // (STEP_DOUBLER_FREQUENCY * 2 kHz - this should never happen)
```

Reply:
I'm happy with having these defines as integer type (will need to change each Platform), I just didn't want to mess with things without the time to fix any unexpected consequences.